auth0/resource_data.go

```diff
@@ -257,13 +258,27 @@ func Float64(d ResourceData, key string, conditions ...Condition) (f *float64) {
 // Bool accesses the value held by key and type asserts it to a pointer to a
 // bool.
 func Bool(d ResourceData, key string, conditions ...Condition) (b *bool) {
+	// We need to keep using this for bool values even if it's marked as deprecated.
+	// See https://github.com/hashicorp/terraform-plugin-sdk/issues/817.
 	v, ok := d.GetOkExists(key)
 	if ok && Any(conditions...).Eval(d, key) {
 		b = auth0.Bool(v.(bool))
 	}
 	return
 }
 
+// Flag accesses the value held by the bool flag key
+// and type asserts it as a pointer to a bool.
+func Flag(configFlags cty.Value, key string) *bool {
+	configFlag := configFlags.GetAttr(key)
+
+	if configFlag.IsNull() {
```

Comment:
Terraform Plugin SDK v2 uses github.com/hashicorp/go-cty/cty under the hood and because of that we are now able to distinguish between values that are set in the config (true/false) and nil values (not present in the config) correctly making sure we also omit them from the PATCH payload if not present in the config.